spikeinterface/core/waveform_extractor.py

```diff
@@ -1424,11 +1427,18 @@ def extract_waveforms(recording, sorting, folder=None,
             return we
     
     if sparse:
-        sparsity = precompute_sparsity(recording, sorting, ms_before=ms_before, ms_after=ms_after,
-                                               num_spikes_for_sparsity=num_spikes_for_sparsity,
-                                               **estimate_kwargs, **job_kwargs)
-    else:
-        sparsity = None
+        if sparsity is None:
+            sparsity = precompute_sparsity(recording, sorting, ms_before=ms_before, ms_after=ms_after,
+                                           num_spikes_for_sparsity=num_spikes_for_sparsity,
+                                           **estimate_kwargs, **job_kwargs)
+        else:
+            assert isinstance(sparsity, ChannelSparsity), "'sparsity' must be a ChannelSparsity object"
+            unit_id_to_channel_ids = sparsity.unit_id_to_channel_ids
+            assert all(u in sorting.unit_ids for u in unit_id_to_channel_ids), \
+                    "Invalid unit ids in sparsity"
+            for channels in unit_id_to_channel_ids.values():
+                assert all(ch in recording.channel_ids for ch in channels), \
+                    "Invalid channel ids in sparsity"
```

Comment:
I think we should have something more easy.
In short `extract_waveforms(recording, sorting, sparsity=something)` should be possible.
Here we force `extract_waveforms(recording, sorting, sparsity=something, sprase=True)`

In my mind `sprse=True` is "try to make make it sparse with estimation"

Reply:
Ok, so if you pass sparsity `sparse` is ignored ;) I agree